src/OperatorMsg.chpl

```diff
@@ -625,420 +386,146 @@ module OperatorMsg
-                        var e = st.addEntry(rname, r.size, real);
-                        e.a = val:real - r.a;
-                    }
-                    when "*" {
-                        var e = st.addEntry(rname, r.size, real);
-                        e.a = val:real * r.a;
-                    }
-                    otherwise {
-                        var errorMsg = notImplementedError(pn,dtype,op,right.dtype);
-                        omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                        return new MsgTuple(errorMsg, MsgType.ERROR); 
-                    }
-                }
+            var e = st.addEntry(rname, r.size, uint);
+            return doBinOpsv(val, r, e, op, dtype, rname, pn, st);
+          }
+          when (DType.UInt64, DType.Int64) {
+            var val = try! value:uint;
+            var r = toSymEntry(right,int);
+            if boolOps.contains(op) {
+              var e = st.addEntry(rname, r.size, bool);
+              return doBinOpsv(val, r, e, op, dtype, rname, pn, st);
+            }
+            // + and - both result in real outputs to match NumPy
+            if op == "+" || op == "-" {
+              var e = st.addEntry(rname, r.size, real);
+              return doBinOpsv(val, r, e, op, dtype, rname, pn, st);
+            } else {
+              // isn't + or -, so we can use LHS to determine type
+              var e = st.addEntry(rname, r.size, uint);
+              return doBinOpsv(val, r, e, op, dtype, rname, pn, st);
             }
-            when (DType.Float64, DType.Bool) {
-                var val = try! value:real;
-                var r = toSymEntry(right, bool);
-                select op {
-                    when "+" {
-                        var e = st.addEntry(rname, r.size, real);
-                        e.a = val + r.a:real;
-                    }
-                    when "-" {
-                        var e = st.addEntry(rname, r.size, real);
-                        e.a = val - r.a:real;
-                    }
-                    when "*" {
-                        var e = st.addEntry(rname, r.size, real);
-                        e.a = val * r.a:real;
-                    }
-                    otherwise {
-                        var errorMsg = notImplementedError(pn,dtype,op,right.dtype);
-                        omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                        return new MsgTuple(errorMsg, MsgType.ERROR); 
-                    }
-                }
+          }
+          when (DType.Int64, DType.UInt64) {
+            var val = try! value:int;
+            var r = toSymEntry(right,uint);
+            if boolOps.contains(op) {
+              var e = st.addEntry(rname, r.size, bool);
+              return doBinOpsv(val, r, e, op, dtype, rname, pn, st);
             }
-            otherwise {
-                var errorMsg = unrecognizedTypeError(pn,
```

Comment:
I think you need some `unrecognizedTypeError` errors in the code instead of the `notImplementedError`
when you reach the end of the type-tuple `select` you should return this kind of error.